Library/Homebrew/cmd/install-vendor.sh

```diff
@@ -0,0 +1,123 @@
+# Hide shellcheck complaint:
+# shellcheck source=/dev/null
+source "$HOMEBREW_LIBRARY/Homebrew/utils/lock.sh"
+
+if [[ -n "$HOMEBREW_OSX" ]]
+then
+  if [[ "$HOMEBREW_PROCESSOR" = "Intel" ]]
+  then
+    ruby_URL="https://github.com/xu-cheng/homebrew-portable/releases/download/20160512/portable-ruby-2.0.0-p648.i386-darwin9.tar.gz"
+    ruby_SHA="035ccdc2c073172148c5ffe3e7004011862b400c6aa3a2831c1cec46a0876de5"
+  else
+    ruby_URL=""
+    ruby_SHA=""
+  fi
+else
+  ruby_URL=""
+  ruby_SHA=""
+fi
+
+cleanup-old() {
+  # Vendor tools will be installed to Library/Homebrew/vendor/<name>/<version>
+  # with a Library/Homebrew/vendor/<name>/opt symlink to point to current version.
+  # This enables us to push updates in the future by simply update the symlink.
+  for version in "$HOMEBREW_LIBRARY/Homebrew/vendor/$VENDOR_NAME"/*
+  do
+    [[ -L "$version" && "$(basename "$version")" = "opt" ]] && continue # skip opt symlink
+    rm -rf "$version"
+  done
+}
+
+install() {
+  local curl_args
+  local tar_args
+  local sha
+
+  if [[ -n "$HOMEBREW_VERBOSE" ]]
+  then
+    curl_args="-fLA"
+    tar_args="xvzf"
+  elif [[ -z "$HOMEBREW_QUIET" ]]
+  then
+    curl_args="-#fLA"
+    tar_args="xzf"
+  else
+    curl_args="-sfLA"
+    tar_args="xzf"
+  fi
+
+  VENDOR_FILE="$(/usr/bin/mktemp "/tmp/homebrew-$VENDOR_NAME.XXXXXX")"
```

Comment:
Maybe just put this in `HOMEBREW_CACHE` instead to avoid unnecessary redownloading if the already downloaded version has the same checksum.